pkg/shell/nav.tsx

```diff
@@ -243,38 +302,26 @@ export function CockpitNavItem(props) {
     );
 }
 
-CockpitNavItem.propTypes = {
-    name: PropTypes.string.isRequired,
-    href: PropTypes.string.isRequired,
-    onClick: PropTypes.func,
-    status: PropTypes.object,
-    active: PropTypes.bool,
-    keyword: PropTypes.string,
-    term: PropTypes.string,
-    header: PropTypes.string,
-    actions: PropTypes.node,
-};
-
-export const PageNav = ({ state }) => {
+export const PageNav = ({ state } : { state: ShellState }) => {
     const {
         current_machine,
         current_manifest_item,
         current_machine_manifest_items,
         page_status,
     } = state;
 
-    if (!current_machine || current_machine.state != "connected") {
+    if (!current_machine || current_machine.state != "connected" || !current_machine_manifest_items || !current_manifest_item) {
```

Comment:
Non-typing behaviour change? Or does that just help with type narrowing?

Reply:
Yes, for narrowing.  This is better written as an assert.

Reply:
Done.